apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1427,6 +1424,31 @@ func (u *UniterAPI) WatchRelationUnits(args params.RelationUnits) (params.Relati
 	return result, nil
 }
 
+// SetRelationStatus updates the status of the specified relations.
+func (u *UniterAPI) SetRelationStatus(args params.RelationStatusArgs) (params.ErrorResults, error) {
+	var statusResults params.ErrorResults
+
+	changeOne := func(arg params.RelationStatusArg) error {
+		rel, err := u.st.Relation(arg.RelationId)
+		if errors.IsNotFound(err) {
+			return common.ErrPerm
+		} else if err != nil {
+			return errors.Trace(err)
+		}
```

Comment:
check here that the relation involves the unit, and return common.ErrPerm if not?

Reply:
also, further to later comments, I think we need to require that the unit is the application leader

Reply:
done